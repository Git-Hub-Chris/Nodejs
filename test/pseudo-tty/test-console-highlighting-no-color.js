'use strict';
require('../common');
const { Console } = require('console');

// Work with no color
process.env.NODE_DISABLE_COLORS = true;
process.env.FORCE_COLOR = 0;

const options = {
  stdout: process.stdout,
  stderr: process.stderr,
  ignoreErrors: true,
  highlight: {
    // Indicators should always work even if no coloring enabled
    warn: { bgColor: 'bgYellow', indicator: '[W]' },
    error: { bgColor: 'bgRed', indicator: '[E]' },
  }
};

const newInstance = new Console(options);
newInstance.warn(new Error('test'));
newInstance.error(new Error('test'));
