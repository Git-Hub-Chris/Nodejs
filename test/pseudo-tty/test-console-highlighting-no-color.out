[W] Error: test
    at * (*test-console-highlighting-no-color.js:*)
*    at *
*    at *
*    at *
*    at *
*    at *
*    at *

[E] Error: test
    at * (*test-console-highlighting-no-color.js:*)
*    at *
*    at *
*    at *
*    at *
*    at *
*    at *
